### ISSUE TITLE
Unofficial depth data API is no longer available for new panoramas

### ISSUE THREAD

**Issue by tongning** [ProjectSidewalk/SidewalkWebpage#520]
With the new Google Maps API update, the unofficial depth data is no longer available for new panorama images, which used to be available at the URL `http://maps.google.com/cbk?output=xml&cb_client=maps_sv&hl=en&dm=1&pm=1&ph=1&renderer=cubic,spherical&v=4&panoid=<panorama_id>`. The depth data is still available for old panoramas, but the new API contains lots of new imagery. Because the depth data is used to render the auditing interface, many panoramas are now showing up gray.

This issue seems to be affecting every project that uses the unofficial API, such as [this game](http://weburbanist.com/2014/03/25/urban-jungle-street-view-3d-hack-uses-hidden-depth-data/) I found. Unfortunately, there does not seem to be an official way to obtain depth data.

**Comment by manaswis** [comment]
Tutorial is broken due to this.

<img width="1277" alt="screen shot 2017-02-20 at 6 08 43 pm" src="https://cloud.githubusercontent.com/assets/2873216/23144730/acb71712-f797-11e6-8e0b-747baaef1b36.png">

**Comment by jonfroehlich** [comment]
Please update this Issue according to @kotarohara's email. He indicated that he was able to find some depth data by sniffing network traffic (but didn't have time to reverse engineer encoding). This Issue should also be marked as highest priority.

**Comment by manaswis** [comment]
Kotaro's response:

>I'm not sure if the depth data is the issue for GSV images not showing up. It is likely that they took down the Street View image for that location. If this is the case, you have to find the image that is nearby the original location then adjust the onboarding script.
>
>I also doubt that Google removed the depth data for the existing panoramas both because (i) it doesn't make sense for Google to do that and (ii) if you go to the Street View interface on Google Maps (e.g., this), it seems like they are still using the depth/point cloud data. Actually what makes you think that they removed the point cloud data? 
>
>After a quick exploration of the network traffic on Google's Street View interface, it seems like they changed the API end points as well as the data format for the depth data (actually, I guess they also changed the end points for the other meta data). The new end point look something like this for this Street View. I think ```data[1][0][5][0][5][3][2]```, something that look like base64 encoded data is the depth (see the screenshot). I'm not sure so you should decode and check. But you definitely have to update the part of the code that queries the meta data. I highly doubt that they changed the encoding method for the depth data.

![depthdata_network_traffic](https://cloud.githubusercontent.com/assets/2873216/23270091/c14ca48c-f9c0-11e6-9d11-40a1c8cd4c8f.png)

**Comment by manaswis** [comment]
During the investigations, we found that the depth data is missing when the imagery (and the associated metadata) for a particular panorama id is not available. This is usually the case when it is replaced with new imagery at a location. #537 shows a different issue - the screen goes blank due to missing imagery while auditing (where the panorama IDs are fetched dynamically). Closing this issue and tracking the missing imagery/depth issue in #537.
